COCOAPODS.md

```diff
@@ -1,150 +1,381 @@
+1. Create an Xcode project with `Podfile` if you haven’t done this yet.
+2. Create a hierarchical structure of your Kotlin projects - a root and child projects.
+3. Add the path to your Xcode project `Podfile` with `podfile = project.file(..)` to `build.gradle.kts` (`build.gradle`) of
+ the root project.  
+    This step helps synchronize your Xcode project with Kotlin Pod dependencies by calling `pod install` for your 
+    `Podfile`.
+4. If the root project doesn't represent a Kotlin Pod, specify that you don’t need a `Podspec` for this project - `noPodspec()`.
 
-```kotlin
-kotlin {
-    cocoapods {
-        // Configure a dependency on AFNetworking.
-        // The CocoaPods version notation is supported.
-        // The dependency will be added to all macOS and iOS targets.
-        pod("AFNetworking", "~> 3.2.0")
+    <div class="sample" markdown="1" theme="idea" data-highlight-only>
+    
+    ```kotlin
+    kotlin {
+        ios()
+        cocoapods {
+            noPodspec()
+            podfile = project.file("ios-app/Podfile")
+        }
     }
-}
-```
+    ```
+    
+    </div>
 
-</div>
+5. Add dependencies to Pod libraries that you want to use to the build script of your child projects using `pod()`.  
+    > Do NOT add a path to `Podfile` in your child projects.
+    {:.note}
+                                                                                      
+    <div class="sample" markdown="1" theme="idea" data-highlight-only>
+    
+    ```kotlin
+    kotlin {
+        ios()
+        cocoapods {
+            summary = "Working with external pods from Kotlin/Native using CocoaPods"
+            homepage = "https://github.com/JetBrains/kotlin-native"
+            pod("AFNetworking", "~> 4.0.0")
+        }
+    }
+    ```
+    
+    </div>
 
-To use these dependencies from a Kotlin code, import a package `cocoapods.<library-name>`.
-In the example above, it's `cocoapods.AFNetworking`.
+6. Add names and paths of the Kotlin Pods you want to include in the Xcode project to `Podfile`.
 
-The dependencies declared in this way are added in the podspec file and downloaded during
-the execution of `pod install`.
+    <div class="sample" markdown="1" theme="idea" mode="ruby" data-highlight-only>
+    
+    ```ruby
+    use_frameworks!
+   
+    platform :ios, '9.0'
+    
+    target 'ios-app' do
+       pod 'firstKotlinPod', :path => '../firstKotlinPod'
+       pod 'secondKotlinPod', :path => '../secondKotlinPod'
+    end
+    ```
+    
+    </div>
 
-> __Important:__ To correctly import the dependencies into the Kotlin/Native module, the
-Podfile must contain either [`use_modular_headers!`](https://guides.cocoapods.org/syntax/podfile.html#use_modular_headers_bang)
-or [`use_frameworks!`](https://guides.cocoapods.org/syntax/podfile.html#use_frameworks_bang)
-directive.
+7. Re-import the project.
+
+>If you get an error about a duplicate symbol when building your Xcode project, add `kotlin.native.cacheKind=none` to 
```

Comment:
The whole advice is not exactly correct.
1. The same issue can have different consequences, not only duplicate symbol errors.
2. AFAIK, using multiple pods almost always gets affected by the issue, so the "if you get an error" part doesn't make much sense, it should be ok to recommend adding the line unconditionally.

Reply:
As we agreed, I've removed the whole scenario.